src/gleam/http.gleam

```diff
@@ -556,3 +556,28 @@ pub fn expire_resp_cookie(resp, name, attributes) {
   let attrs = CookieAttributes(..attributes, max_age: Some(0))
   set_resp_cookie(resp, name, "", attrs)
 }
+
+/// Get the origin request header
+///
+/// If no "Origin" header is found in the request, falls back to the "Referer"
+/// header.
+pub fn get_req_origin(req: Request(body)) -> Option(String) {
+  case get_req_header(req, "origin") {
+    Ok(origin) -> Some(origin)
+    Error(Nil) ->
+      case get_req_header(req, "referer") {
+        Ok(ref) ->
+          case ref
+          |> uri.parse {
+            Ok(uri) ->
+              Some(
+                option.unwrap(uri.scheme, "")
```

Comment:
I don't think it's the right thing to unwrap these options. if the referrer header doesn't have a scheme or host returning Some("://") is a bad response. I think if the referrer is missing an origin then None should be returned.

Probably worth writing a test case for these

Reply:
Using 'option.from_result` now. Also added test cases for if the referrer is missing the scheme or host.